cpu/riscv_common/include/pmp.h

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2023 Bennet Blischke <bennet.blischke@haw-hamburg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup         cpu_riscv_common
+ * @{
+ *
+ * @file
+ * @brief           RISC-V PMP configuration options
+ *
+ * RISCV implementations using this peripheral must define the `NUM_PMP_ENTRIES`
+ * `NUM_PMP_ENTRIES` must be 16 or 64.
+ *
+ * @author          Bennet Blischke
+ */
+
+#ifndef PMP_H
+#define PMP_H
+
+#include <stdint.h>
+#include "cpu_conf.h"
+#include "vendor/riscv_csr.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define CSR_PMPCFG0 0x3a0
+#define CSR_PMPCFG1 0x3a1
+#define CSR_PMPCFG2 0x3a2
+#define CSR_PMPCFG3 0x3a3
+#define CSR_PMPADDR0 0x3b0
+#define CSR_PMPADDR1 0x3b1
+#define CSR_PMPADDR2 0x3b2
+#define CSR_PMPADDR3 0x3b3
+#define CSR_PMPADDR4 0x3b4
+#define CSR_PMPADDR5 0x3b5
+#define CSR_PMPADDR6 0x3b6
+#define CSR_PMPADDR7 0x3b7
+#define CSR_PMPADDR8 0x3b8
+#define CSR_PMPADDR9 0x3b9
+#define CSR_PMPADDR10 0x3ba
+#define CSR_PMPADDR11 0x3bb
+#define CSR_PMPADDR12 0x3bc
+#define CSR_PMPADDR13 0x3bd
+#define CSR_PMPADDR14 0x3be
+#define CSR_PMPADDR15 0x3bf
+
+#define PMP_NONE    0x00
+#define PMP_R       0x01
+#define PMP_W       0x02
+#define PMP_X       0x04
+
+#define PMP_A       0x18
+#define PMP_OFF     0x00
+#define PMP_TOR     0x08
+#define PMP_NA4     0x10
+#define PMP_NAPOT   0x18
+
+#define PMP_L       0x80
+
+#define _WRITE_PMPCFG(REG, VALUE) write_csr(REG, VALUE)
+#define _READ_PMPCFG(REG) read_csr(REG)
```

Comment:
Extra step needed to enforce evaluation of macro.

Reply:
better add a comment